src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterClusterStateUpdateStatsCollectorTests.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.ESResources;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.config.PerformanceAnalyzerController;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.config.overrides.ConfigOverridesWrapper;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsConfiguration;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader_writer_shared.Event;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import junit.framework.TestCase;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.cluster.service.MasterService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ESResources.class})
+public class MasterClusterStateUpdateStatsCollectorTests extends TestCase {
+    ObjectMapper mapper = new ObjectMapper();
+    @Test
+    public void testMasterClusterStateUpdateStats_saveMetricValues() {
+        MetricsConfiguration.CONFIG_MAP.put(MasterClusterStateUpdateStatsCollector.class, MetricsConfiguration.cdefault);
+        System.setProperty("performanceanalyzer.metrics.log.enabled", "False");
+        long startTimeInMills = 1153721339;
+        PerformanceAnalyzerController controller = Mockito.mock(
+                PerformanceAnalyzerController.class);
+        ConfigOverridesWrapper configOverrides = Mockito.mock(ConfigOverridesWrapper.class);
+        MasterClusterStateUpdateStatsCollector masterClusterStateUpdateStatsCollector = new MasterClusterStateUpdateStatsCollector(
+                controller, configOverrides);
+        Mockito.when(controller.isCollectorEnabled(configOverrides, "MasterClusterStateUpdateStatsCollector"))
+                .thenReturn(true);
+        masterClusterStateUpdateStatsCollector.saveMetricValues("master_cluster_update",
+                startTimeInMills);
+        List<Event> metrics =  new ArrayList<>();
+        PerformanceAnalyzerMetrics.metricQueue.drainTo(metrics);
```

Comment:
Can we clear metricQueue before running UT? otherwise these UT might become flaky.
Sample PR: https://github.com/opendistro-for-elasticsearch/performance-analyzer/pull/300/files

Reply:
done